### PR TITLE
rpc: add node color to NodeUpdate and GetInfo

### PR DESCRIPTION
**Pull request by xsb**
This adds the node color to both `NodeUpdate` (as part of a`GraphTopologyUpdate`) and to `GetInfo` response. Tested with a Go gRPC client.

Fixes https://github.com/lightningnetwork/lnd/issues/2255 and https://github.com/lightningnetwork/lnd/issues/2268

### CODE REVIEW

File: channeldb/graph.go [hunk 1]
```diff
@@ -1752,6 +1752,11 @@ func (l *LightningNode) PubKey() (*btcec.PublicKey, error) {
 	return key, nil
 }
 
+// ColorHexCode returns the node's color in hex code format.
+func (l *LightningNode) ColorHexCode() string {
```
Comment (halseth) [inline channeldb/graph.go]: Better to make this a utility method like `parseHexColor` I think

Comment (xsb) [reply channeldb/graph.go]: sure, makes sense

File: server.go [hunk 2]
```diff
@@ -2842,6 +2841,11 @@ func parseHexColor(colorStr string) (color.RGBA, error) {
 	return color.RGBA{R: colorBytes[0], G: colorBytes[1], B: colorBytes[2]}, nil
 }
 
+// encodeHexColor takes a color and returns it in hex code format.
+func encodeHexColor(color color.RGBA) string {
```
Comment (halseth) [inline server.go]: with this addition, the `server_test.go` should be expended to test this!

File: routing/notifications.go [hunk 3]
```diff
@@ -312,6 +316,7 @@ func addToTopologyChange(graph *channeldb.ChannelGraph, update *TopologyChange,
 			Addresses:   m.Addresses,
 			IdentityKey: pubKey,
 			Alias:       m.Alias,
+			Color:       encodeHexColor(m.Color),
```
[the remark this answers is not in the document]
Comment (xsb) [reply routing/notifications.go]: This is needed to implement https://github.com/lightningnetwork/lnd/issues/2255

[the remark this answers is not in the document]
Comment (halseth) [reply routing/notifications.go]: Good point! Ideally, the `encode` method would be shared, and not duplicated across the two packages though...

**Comment by xsb** [comment]
@halseth I made it with a separate commit per package as you suggested and also added a test. Something I am not convinced though is exposing the utility method EncodeHexColor from the routing package but I wasn't sure where to put it. Seems weird to me, I am open to changes.

**Comment by xsb** [comment]
Rebased

**Comment by xsb** [comment]
@halseth rebased with latest master + proto files regenerated

**Comment by cfromknecht** [comment]
@xsb looks like we have another conflict 😶

**Comment by xsb** [comment]
@halseth @cfromknecht Rebased again.

Note: While regenerating the proto files I got a different result for `lnrpc/invoicesrpc/invoices.pb.go` that I ignored it for the sake of cleanliness. It seems to come from the recently merged #3019. Either the author of that PR or myself are not running the correct protoc, etc. Probably not super important but better to clarify before merging this.

**Comment by halseth** [comment]
Yet another proto conflict 🤕 

> Note: While regenerating the proto files I got a different result for lnrpc/invoicesrpc/invoices.pb.go that I ignored it for the sake of cleanliness.

If you still see this after rebasing on master, check your proto+tools version. I just tried rebasing this on master, and regenerating the protos doesn't create any changes to that file for me.

**Comment by xsb** [comment]
@halseth rebased again. No issue regenerating protos this time (tried on master and on my branch).

**Comment by cfromknecht** [comment]
sorry @xsb, there was a conflict w/ #3082 :/ you'll need to rebase on master and ensure that your protogen version is on 0.1.3 (step 2 of https://github.com/lightningnetwork/lnd/tree/master/lnrpc#generate-protobuf-definitions)

Feel free to ping me when you're ready, hoping to minimize the number of rebases on your end 😬

**Comment by xsb** [comment]
@cfromknecht  rebased again after upgrading to new protobuf version

**Pull request merged**